core/app/models/spree/order.rb

```diff
@@ -535,7 +535,30 @@ def create_proposed_shipments
         raise CannotRebuildShipments.new(I18n.t('spree.cannot_rebuild_shipments_shipments_not_pending'))
       else
         shipments.destroy_all
-        self.shipments = Spree::Config.stock.coordinator_class.new(self).shipments
+
+        new_shipments = Spree::Config.stock.coordinator_class.new(self).shipments
+        # By default, rails saves automatically all the new objects recursively
+        # whenever the shipments are assigned to the order, but in terms of inventory
+        # units, it inserts them one by one, we can take advantage of new Rails 6
+        # insert_all functionality here
+        if Spree::InventoryUnit.respond_to?(:insert_all)
+          self.shipments = new_shipments.map do |shipment|
+            # We need to save temporarily the inventory unit attributes and
+            # unset  them from shipment so they do not get saved
+            inventory_units_attributes = shipment.inventory_units.map(&:value_attributes)
+            shipment.inventory_units = []
+            shipment.save
+            # Then insert them all at once
+            if inventory_units_attributes.size.positive?
+              Spree::InventoryUnit.insert_all(
```

Comment:
I haven't played with `insert_all` yet. Are you able to do the `insert_all` direction on the association to avoid the map?

```suggestion
              shipment.inventory_units.insert_all(inventory_units_attributes)
```

Reply:
I've literally just tried that locally and it doesn't seem to work, unfortunately.